component/block_cache/block_cache.go

```diff
@@ -379,6 +378,20 @@ func (bc *BlockCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Han
 	return handle, nil
 }
 
+// validateBlockList: Validates the blockList.
+// Each Block's size must equal to blockSize set in config and last block size <= config's blockSize
+// returns true, if blockList is valid
+func (bc *BlockCache) validateBlockList(fileName string, blockSize uint64, blockList *internal.CommittedBlockList) bool {
+	listLen := len(*blockList)
+	for idx, block := range *blockList {
+		if (idx < (listLen-1) && block.Size != blockSize) || (idx == (listLen-1) && block.Size > blockSize) {
+			log.Err("BlockCache::OpenFile : Block size mismatch for %s [block: %v, size: %v]", fileName, block.Id, block.Size)
```

Comment:
In log we print the class name and function name. here function name shall be used instead of "OpenFile"